src/wallet.js

```diff
@@ -500,14 +500,36 @@ const wallet = {
    * @inner
    */
   getAddressToUse() {
-    const address = storage.getItem('wallet:address');
+    const address = this.getCurrentAddress();
     // Updating address because the last one was used
+    this.updateCurrentAddress();
+    return address;
+  },
+
+  /**
+   * Get current address
+   *
+   * @return {string} address
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getCurrentAddress() {
+    return storage.getItem('wallet:address');
+  },
+
+  /**
+   * Update current address, if has not yet reached the GAP_LIMIT
```

Comment:
Suggestion:

Move to the next address in the derivation chain and return it.

It may not move to the next address if the number of unused addresses has reached the GAP_LIMIT. In this case, it returns the same as `getCurrentAddress`.